integration-tests/manifests/annotations/validate_annotations_test.go

```diff
@@ -0,0 +1,1003 @@
+package annotations
```

Comment:
For future reference, I suggest breaking code like this into separate PRs especially when we know they are separated test cases.

In the description we have 4 main use cases then I'm assuming this PR can be broken into 4 PRs. 1000 lines of code review is difficult to track of.

Reply:
nit: Idk if all the tests are dependent on each other. but if not it would be better to divide into `Deployment`, `Daemonset`,`Namespace`,and `Statefulset`, so something like `validate_annotations_deployment_test.go`,`validate_annotations_daemonset_test.go`, etc

This way when making a test change for a specific k8s component, it's more clear instead of having to scroll through other unrelated tests.